app/src/main/java/com/example/themealsapp/presentation/viewmodel/MealsViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.themealsapp.presentation.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.themealsapp.domain.model.Meal
+import com.example.themealsapp.domain.use_case.GetMealsByName
+import com.example.themealsapp.utils.UIState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+
+private const val TAG = "MealsViewModel"
+@HiltViewModel
+class MealsViewModel @Inject constructor(
+    private val getMealsByName: GetMealsByName
+): ViewModel(){
+
+    init {
+        onSeach("")
```

Comment:
lets check the method name, and the argument why is it empty string?

Reply:
changed to "onSearchMealsByName()"